docs/reference/aggregations/metrics/t-test-aggregation.asciidoc

```diff
@@ -0,0 +1,111 @@
+[role="xpack"]
+[testenv="basic"]
+[[search-aggregations-metrics-ttest-aggregation]]
+=== TTest Aggregation
+
+A `t_test` metrics aggregation that performs a statistical hypothesis test in which the test statistic follows a Student's t-distribution
+under the null hypothesis on numeric values extracted from the aggregated documents or generated by provided scripts.
+
```

Comment:
Should perhaps add a layman explanation afterwards?  Maybe something like "In practice, this will tell you if the difference between two population means are statistically significant", or something like that?